code/go/internal/validator/semantic/validate_hbs_templates_test.go

```diff
@@ -0,0 +1,144 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package semantic
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/package-spec/v3/code/go/internal/fspath"
+)
+
+func TestValidateTemplateDir(t *testing.T) {
+	t.Run("empty directory", func(t *testing.T) {
+		tmpDir := t.TempDir()
+		pkgDir := path.Join(tmpDir, "package")
```

Comment:
Nit. Most paths in these tests are native paths, they should be joined with `filepath.Join`.